ftl/common/builder_runner.py

```diff
@@ -55,25 +57,23 @@ def __init__(self, args, builder, cache_version_str):
             mount=[self.base_name])
         self.builder = builder.From(self.ctx)
 
-    def GetCacheKey(self, descriptor_files):
-        descriptor = None
-        for f in descriptor_files:
-            if self.ctx.Contains(f):
-                descriptor = f
-                descriptor_contents = self.ctx.GetFile(descriptor)
-                break
-        if not descriptor:
-            logging.info('No package descriptor found. No packages installed.')
+    def GetCacheKey(self, key):
+        if key is None:
             return None
-        return hashlib.sha256(descriptor_contents).hexdigest()
+        return hashlib.sha256(key).hexdigest()
+
+    def GetCacheKeys(self, keys):
+        if keys[0] is None:
```

Comment:
Same question here. Also, what if keys itself is none, or length 0?

Reply:
It should definitely be done better, currently I was using 'None' as a sentinel to reconcile that for a builder with no cache keys (being that the app has not requirements), we still want to use a cached base image.  It should be split up differently but that is what the 'None' is for currently